source/lac/trilinos_vector.cc

```diff
@@ -504,6 +505,24 @@ namespace TrilinosWrappers
       const Epetra_MpiComm *v_comm = dynamic_cast<const Epetra_MpiComm *>(&v.vector->Comm());
       const bool same_communicators = my_comm != NULL && v_comm != NULL &&
                                       my_comm->DataPtr() == v_comm->DataPtr();
+      // Need to ask MPI whether the communicators are the same. We would like
+      // to use these checks but currently we cannot make sure the memory of
+      // my_comm is not stale from some MPI_Comm_free somewhere. Thus, the
+      // following code is commented out.
```

Comment:
can you specifically reference `GrowingVectorMemory` to make clear what use case you have in mind?